lib/collection/src/operations/types.rs

```diff
@@ -984,6 +984,18 @@ impl CollectionError {
             Self::ObjectStoreError { .. } => false,
         }
     }
+
+    pub fn is_pre_condition_failed(&self) -> bool {
+        matches!(self, Self::PreConditionFailed { .. })
+    }
+
+    pub fn is_missing_point(&self) -> bool {
+        match self {
+            CollectionError::NotFound { what } => what.contains("No point with id"),
```

Comment:
I would prefer some strict type checking here, but I understand that may not really be possible right now.

Reply:
Yeah, too much refactoring to propagate this error properly. :/